staging/src/k8s.io/pod-security-admission/policy/check_allowPrivilegeEscalation.go

```diff
@@ -84,6 +84,11 @@ func allowPrivilegeEscalation_1_8(podMetadata *metav1.ObjectMeta, podSpec *corev
 }
 
 func allowPrivilegeEscalation_1_25(podMetadata *metav1.ObjectMeta, podSpec *corev1.PodSpec) CheckResult {
+	// See KEP-127: https://github.com/kubernetes/enhancements/blob/308ba8d/keps/sig-node/127-user-namespaces/README.md?plain=1#L411-L447
```

Comment:
Can you link to the part of the user namespaces design that describes how privilege escalation is mitigated by user namespaces. runAsUser and runAsNonRoot are much more clearly mitigated by the uid mapping.

Reply:
There is on actual documentation about privilege escalation but I extended the comment to clarify how it works.

Reply:
Dropping runAsNonRoot and runAsUser restrictions for user namespace pods makes sense, those changes lgtm.

I spent some time talking through this change with @mtaufen and @vinayakankugoyal, and I think disallowing privilege escalation and requiring dropping capabilities still provides significant protections, even with user namespaces. Allowing arbitrary capabilities exposes many more syscalls and kernel code paths, and I don't think typical vulnerabilities (buffer overflows, etc) are mitigated by a uid mapping. @vinayakankugoyal can provide more details. Before we relax privilege escalation and arbitrary capabilities for user namespace pods, we need to be confident we're not ending up in a less secure state.

If we get the feature gate plumbing straightened out (https://github.com/kubernetes/kubernetes/pull/118760/files#r1373276402) and scope this down to the runAsUser / mustRunAsNonRoot checks, that works for now.

Reply:
That sounds reasonable for now. We can revisit capabilities later.

Reply:
Sounds good to me.